app/src/test/java/edu/wpi/first/shuffleboard/app/json/DashboardTabPaneSaverTest.java

```diff
@@ -1,66 +1,133 @@
+import edu.wpi.first.shuffleboard.api.sources.DataSource;
+import edu.wpi.first.shuffleboard.api.widget.Component;
+import edu.wpi.first.shuffleboard.api.widget.Components;
+import edu.wpi.first.shuffleboard.api.widget.Widget;
+import edu.wpi.first.shuffleboard.api.widget.WidgetType;
+import edu.wpi.first.shuffleboard.app.DashboardData;
 import edu.wpi.first.shuffleboard.app.components.DashboardTab;
 import edu.wpi.first.shuffleboard.app.components.DashboardTabPane;
-import edu.wpi.first.shuffleboard.api.util.AsyncUtils;
-import edu.wpi.first.shuffleboard.api.util.FxUtils;
-import edu.wpi.first.shuffleboard.api.util.NetworkTableUtils;
+import edu.wpi.first.shuffleboard.app.components.Tile;
 
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.testfx.framework.junit5.ApplicationTest;
 
 import java.io.InputStreamReader;
 import java.io.Reader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
 
-import javafx.stage.Stage;
+import javafx.beans.property.Property;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.layout.Pane;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@Disabled("Depends on network tables")
 public class DashboardTabPaneSaverTest extends ApplicationTest {
 
-  @Override
-  public void start(Stage stage) throws Exception {
-    // Just here so we can run on the FX thread
-  }
-
-  @BeforeEach
-  public void setUp() {
-    NetworkTableUtils.shutdown();
-    NetworkTableInstance inst = NetworkTableInstance.getDefault();
-    inst.setUpdateRate(0.01);
-    AsyncUtils.setAsyncRunner(Runnable::run);
-    inst.getEntry("/LiveWindow/TestSystem/Ultrasonic/Value").setDouble(0.5);
-    inst.getEntry("/LiveWindow/TestSystem/Compass/Value").setDouble(0.5);
-    inst.getEntry("/LiveWindow/Elevator/p").setDouble(0.5);
-    inst.getEntry("/LiveWindow/Elevator/d").setDouble(0.5);
-    inst.getEntry("/LiveWindow/Elevator/f").setDouble(0.5);
-
-    inst.getEntry("/LiveWindow/~STATUS~/LW Enabled").setBoolean(true);
-    inst.getEntry("/LiveWindow/Elevator/enabled").setBoolean(false);
-    inst.waitForEntryListenerQueue(-1.0);
+  @BeforeAll
+  public static void setup() {
+    Components components = new Components();
+    Components.setDefault(components);
+    components.register(new DummyBooleanBoxType());
   }
 
-  @AfterEach
-  public void tearDown() {
-    AsyncUtils.setAsyncRunner(FxUtils::runOnFxThread);
-    NetworkTableUtils.shutdown();
+  @AfterAll
+  public static void tearDown() {
+    Components.setDefault(new Components());
```

Comment:
Generally, you shouldn't need to do this sort of thing in well decoupled software.